crescent/ext/tasks/task.py

```diff
@@ -0,0 +1,63 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from asyncio import ensure_future, get_event_loop
+from typing import Any, Awaitable, Callable, Sequence, TypeVar
+
+from crescent.bot import Bot
+from crescent.internal.meta_struct import MetaStruct
+
+TaskCallbackT = Callable[[], Awaitable[None]]
+
+__all__: Sequence[str] = ("TaskCallbackT",)
+
+
+class _Task(ABC):
+    def __init__(self, callback: TaskCallbackT) -> None:
+        self.event_loop = get_event_loop()
+        self.callback = callback
+        self.running = True
+        self.app: Bot | None = None
+
+    def start(self) -> None:
+        ensure_future(self._start_inner())
+
+    async def _start_inner(self) -> None:
+        assert self.app is not None
+
+        await self.app.started.wait()
+
+        self.call_next()
+
+    def stop(self) -> None:
+        self.running = False
+
+    def call_async(self) -> None:
+        if not self.running:
+            return
+        ensure_future(self.callback())
+        self.call_next()
```

Comment:
Should this maybe raise an error if self.running is false? Anywhere else where it's called while it may still be non-running can just suppress the error.

Reply:
The idea is that when you stop it, it will not do the next call.
Just realized this has a bug if you stop then start a loop though. Will fix.